src/core/store.ts

```diff
@@ -898,3 +898,22 @@ export const createStore = (
 }
 
 export type Store = ReturnType<typeof createStore>
+
+export const createStoreForExport = (
+  initialValues?: Iterable<readonly [AnyAtom, AnyAtomValue]>
+) => {
+  const store = createStore(initialValues)
+  return {
+    get: <Value>(atom: Atom<Value>): ResolveType<Value> | undefined => {
+      const atomState = store[READ_ATOM](atom)
+      return 'v' in atomState ? atomState.v : undefined
```

Comment:
This is an unfortunate erasure if `type Value = OtherThing | undefined` but the atom can also have a suspense promise. Perhaps the result of `get` can be a more interesting type like `{ value: Value } | undefined`?

Reply:
And, what about the error case? Perhaps that should still throw, although I feel pretty strongly that `store.get(atom)` should never throw the suspense promise.

Reply:
Yeah, this is unfortunate.
I think it should throw on error, and maybe suspense too.
We could have `has` method, hmm.

Reply:
Does `has` cause recompute?

Reply:
Re: throwing promise: I think outside of React library authors it’s pretty unusual to think about throwing promises, whereas throwing errors is common for most JS styles. Given that the public `store` intends to expose Jotai for use *outside* React, I think it might be inappropriate to throw the suspense promise since that’s a kind of react specific, framework internal practice.  But I am behind the times in terms of React best practices so maybe this is better known in the cutting edge JS community?

Reply:
I understand the goal of Jotai is to be as close to React behavior as possible, but maybe this interface is the exception?

Reply:
Hm, probably we can just return a promise.

Reply:
but this isn't nice in terms of typing...

Reply:
281a617 would it be too complicated to explain?

Reply:
> Does `has` cause recompute?

Good point, yeah...

Reply:
https://github.com/pmndrs/jotai/commit/281a617d1247173fb0c4abf00b03ffbbf95f4cf5  This is very tricky to design!

the way I would write this in my personal, typescript only world would be to use a discriminating union result type that always forces the caller to consider the different types of result:

`type GetResult<Value> = { type: ‘value’, value: Resolve<Value> } | { type: ‘suspended’, promise: Promise<???> } | { type: ‘error’, error: unknown }`. 

But this style can be hard for JS-only consumers to understand or get right, and it’s not very natural in JS to return the error. So you could always throw the error, or return the two other variants. 

i am starting to think I am unqualified to understand the suspense cases though, I don’t know enough about suspense in React or how an atom ends up “suspended”.

Reply:
From the typescript perspective, you can't know if an atom will suspend or not.
Do we want to check type=value, even if we never use async in our code?
So, after all, if we were to explain it to someone who don't know suspense (which is likely), the original `return undefined` idea would be easier (we throw on error). We would be simply explaining when the value is not ready, it returns `undefined`. It's like how `Map.get` is typed.

Reply:
So, this: 4fd5afa

Reply:
What about a second method with the result type, for callers that are working with async? You had as an option overload before, that works too. I do think that returning undefined is better than throwing promise.

Reply:
Should the public interface be expressive enough to allow an alternative implementation of useAtom by consumers, including suspense, if they chose to have those semantics? It’s not important to me, though. Another thought exercise about the purpose of the public store API.

Reply:
Yeah, that's the purpose of the RFC issue.
Exposing everything / having a second method would be all possible, but I don't think it's very good developer experience.

281a617 is good in a sense that `unstable_promise` is already introduced in getter in `atom.write`. It's still very unstable though.